### PR TITLE
Don't show "Update DB" notice, when we're already updating the DB.

### PR DESCRIPTION
**Pull request by bobdenotter**
As pointed out by @EJanuszewski in #4899 

> The database message also stays popped up after updating it(on the database update page), I might open a separate issue for this.

This PR fixes that, as soon as #4912 is also fixed.

### CODE REVIEW

**Comment by bobdenotter** [comment]
xzibit

File: src/EventListener/StorageEventListener.php [hunk 1]
```diff
@@ -120,7 +121,12 @@ protected function schemaCheck(GetResponseEvent $event)
         $session = $event->getRequest()->getSession();
         $validSession = $session->isStarted() && $session->get('authentication');
         $expired = $this->schemaManager->isCheckRequired();
-        if ($validSession && $expired && $this->schemaManager->isUpdateRequired()) {
+
+        // Don't show the check if we're in the dbcheck already.
+        $app = ResourceManager::getApp();
```
Comment (GawainLynch) [inline src/EventListener/StorageEventListener.php]: That is gone very soon… like **very** soon… like 3.1.0 soon.

File: src/EventListener/StorageEventListener.php [hunk 2]
```diff
@@ -120,7 +121,12 @@ protected function schemaCheck(GetResponseEvent $event)
...
+        $notInCheck = !in_array($app['request']->get('_route'), ['dbcheck', 'dbupdate_result', 'dbupdate']);
```
Comment (GawainLynch) [inline src/EventListener/StorageEventListener.php]: Kernel event listener event object have the request object

**Comment by bobdenotter** [comment]
@GawainLynch Thanks! Much nicer this way.

**Pull request merged**